lib/rust/ensogl/core/src/system/js/app.rs

```diff
@@ -36,6 +36,18 @@ pub mod js_bindings {
         #[wasm_bindgen(method)]
         #[wasm_bindgen(js_name = registerSetShadersRustFn)]
         pub fn register_set_shaders_rust_fn(this: &App, closure: &Closure<dyn FnMut(JsValue)>);
+
+        /// Show a spinner covering the whole viewport.
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(method)]
+        #[wasm_bindgen(js_name = showProgressIndicator)]
+        pub fn show_progress_indicator(this: &App, progress: f32);
+
+        /// Hide a spinner.
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(method)]
+        #[wasm_bindgen(js_name = hideProgressIndicator)]
+        pub fn hide_progress_indicator(this: &App);
```

Comment:
Somewhere we hide the progress indicator after project load. Can we use this method there? I think I didn't see it in this PR.

Reply:
We can't because it is the responsibility of the JS code that loads WASM. Spinner is tracking download progress and is hidden after it is done automatically.

Reply:
We CAN reuse the spinner if we want. Its API is exposed by JS app that we are connected to in Rust. I can help with it if we want it.

Reply:
@wdanilo what do you mean? We *are* using a spinner from JS app, I was just forced to add a bit of code to make it work from Rust (`show_progress_indicator` and `hide_progress_indicator` methods).